cuda_bindings/docs/source/overview.md

```diff
@@ -291,22 +290,286 @@ nsys profile -s none -t cuda --stats=true <executable>
   - 1080ms
 ```
 
-CUDA Python is also compatible with [NVIDIA Nsight
+`cuda.bindings` is also compatible with [NVIDIA Nsight
 Compute](https://developer.nvidia.com/nsight-compute), which is an
 interactive kernel profiler for CUDA applications. It allows you to have
-detailed insights into kernel performance. This is useful when you’re trying to
+detailed insights into kernel performance. This is useful when you're trying to
 maximize performance ({numref}`Figure 1`).
 
-```{figure} _static/images/Nsigth-Compute-CLI-625x473.png
+```{figure} _static/images/Nsight-Compute-CLI-625x473.png
 :name: Figure 1
 
-Screenshot of Nsight Compute CLI output of CUDA Python example.
+Screenshot of Nsight Compute CLI output of `cuda.bindings` example.
+```
+
+## Preparing kernel arguments
+
+The `cuLaunchKernel` API bindings retain low-level CUDA argument preparation requirements:
+
+* Each kernel argument is a `void*` (i.e. pointer to the argument)
+* `kernelParams` is a `void**` (i.e. pointer to a list of kernel arguments)
+* `kernelParams` arguments are in contiguous memory
+
+These requirements can be met with two different approaches, using either NumPy or ctypes.
+
+### Using NumPy
+
+NumPy [Array objects](https://numpy.org/doc/stable/reference/arrays.html) can be used to fulfill each of these conditions directly.
+
+Let's use the following kernel definition as an example:
+```python
+kernel_string = """\
+typedef struct {
+    int value;
+} testStruct;
+
+extern "C" __global__
+void testkernel(int i, int *pi,
+                float f, float *pf,
+                testStruct s, testStruct *ps)
+{
+    *pi = i;
+    *pf = f;
+    ps->value = s.value;
+}
+"""
+```
+
+The first step is to create array objects with types corresponding to your kernel arguments. Primative Numpy types have the following corresponding kernel types:
```

Comment:
typo: **Primitive** NumPy types

NumPy is usually written with a capital P

Reply:
```suggestion
The first step is to create array objects with types corresponding to your kernel arguments. Primitive NumPy types have the following corresponding kernel types:
```